packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -95,3 +95,144 @@ class HttpRouteWholePathMatch extends HttpRoutePathMatch {
+   *  To disable automatic rewrite, provide `''`.
+   */
+  public static startsWith(prefixPathMatch: string, rewriteTo: string = '/'): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePrefixPathMatch(prefixPathMatch, rewriteTo);
+  }
+
+  /**
+   * The value of the path must match the specified value exactly.
+   * The provided `path` must start with the '/' character.
+   *
+   * @param path the exact path to match on
+   * @param rewriteTo the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static exactly(path: string, rewriteTo?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ exact: path }, { exact: rewriteTo });
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   *
+   * @param regex the regex used to match the path
+   * @param rewriteTo the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static regex(regex: string, rewriteTo?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ regex }, { exact: rewriteTo });
+  }
+
+  /**
+   * Returns the gateway route path match configuration.
+   */
+  public abstract bind(scope: Construct): HttpGatewayRoutePathMatchConfig;
+}
+
+class HttpGatewayRoutePrefixPathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly prefixPathMatch: string,
+    private readonly rewriteTo: string,
+  ) {
+    super();
+
+    if (this.prefixPathMatch[0] !== '/') {
+      throw new Error('Prefix path for the match must start with \'/\', '
+        + `got: ${this.prefixPathMatch}`);
+    }
+
+    if (this.rewriteTo !== '/' && this.rewriteTo !== '') {
+      if (this.prefixPathMatch[this.prefixPathMatch.length-1] !== '/') {
+        throw new Error('When prefix path for the rewrite is specified, prefix path for the match must end with \'/\', '
+          + `got: ${this.prefixPathMatch}`);
+      }
+      if (this.rewriteTo[0] !== '/' || this.rewriteTo[this.rewriteTo.length-1] !== '/') {
+        throw new Error('Prefix path for the rewrite must start and end with \'/\', '
+          + `got: ${this.rewriteTo}`);
+      }
+    }
+  }
+
+  bind(_scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      prefixPathMatch: this.prefixPathMatch,
+      prefixPathRewrite: this.rewriteTo === '/' ? undefined : {
+        defaultPrefix: this.rewriteTo === '' ? 'DISABLED' : undefined,
+        value: this.rewriteTo === '' ? undefined : this.rewriteTo,
+      },
+    };
+  }
+}
+
+class HttpGatewayRouteWholePathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly wholePathMatch: CfnGatewayRoute.HttpPathMatchProperty,
+    private readonly wholePathRewrite?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty,
+  ) {
+    super();
+
+    if (this.wholePathMatch.exact && this.wholePathMatch.exact[0] !== '/') {
```

Comment:
General comment everywhere in this file: you don't have to keep `this.` in the constructor - the values are available as parameters of the constructor.

Reply:
Addressed